Mathlib/GroupTheory/SpecificGroups/Coxeter.lean

```diff
@@ -0,0 +1,461 @@
+/-
+Copyright (c) 2024 Newell Jensen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Newell Jensen
+-/
+import Mathlib.Data.ENat.Basic
+import Mathlib.Data.Matrix.Notation
+import Mathlib.GroupTheory.OrderOfElement
+import Mathlib.GroupTheory.PresentedGroup
+import Mathlib.LinearAlgebra.Matrix.Symmetric
+
+/-!
+# Coxeter Systems
+
+This file defines Coxeter systems and Coxeter groups.
+
+A Coxeter system is a pair `(W, S)` where `W` is a group generated by a set of
+reflections (involutions) `S = {s₁,s₂,...,sₙ}`, subject to relations determined
+by a Coxeter matrix `M = (mᵢⱼ)`.  The Coxeter matrix is a symmetric matrix with entries
+`mᵢⱼ` representing the order of the product `sᵢsⱼ` for `i ≠ j` and `mᵢᵢ = 1`.
+
+When `(W, S)` is a Coxeter system, one also says, by abuse of language, that `W` is a
+Coxeter group.  Coxeter group `W` is defined by the presentation
+`W = ⟨s₁,s₂,...,sₙ | ∀ i j, (sᵢsⱼ)^mᵢⱼ = 1⟩`, where `1` is the identity element of `W`.
```

Comment:
Minor nitpicks:
```suggestion
Coxeter group. The Coxeter group `W` is defined by the presentation
`W = ⟨s₁,s₂,...,sₙ | ∀ i j, (sᵢsⱼ)^mᵢⱼ = 1⟩`, where `1` is the identity element of `W`.
```
Or maybe 'A Coxeter group' ...? Also, is it 'defined' or 'determined' by the presentation?

Reply:
I like 'A Coxeter group' and I would agree that using 'determined' is more accurate since a presentation is not unique.  I like being as correct as possible so thanks for the nits!